### PR TITLE
Deterministic algorithm for invitation acceptance

### PR DESCRIPTION
**Pull request by alexanderjarvis**
As recommended in WWDC 2014 session 709. Use the hash value of the
MCPeerID object and whoever has the greater hash value can accept the
invitation.

http://asciiwwdc.com/2014/sessions/709?q=multipeer

### CODE REVIEW

**Comment by alexanderjarvis** [comment]
Actually just testing this and it looks like it invites itself possible because there are two MCSession's. One in the advertiser and one in the browser.

**Comment by nickswalker** [comment]
I have noticed the problems that this pull request addresses. @jpsim, please merge this in.

**Comment by alexanderjarvis** [comment]
There is another improvement that I am currently doing at a higher level – which is to stop/start the transceiver and create a new session whenever there is a disconnect and zero connected clients.

This is because currently the PeerKit code stops advertising (if it accepts an invitation). This makes sense if there is a cluster of PeerKit devices for perhaps only one to advertise, but maybe all devices should always be advertising. Contrary to the advice found on the interwebs.

File: PeerKit/PeerKit.swift [hunk 1]
```diff
@@ -44,7 +44,7 @@ func didConnecting(myPeerID: MCPeerID, peer: MCPeerID) {
 
 func didConnect(myPeerID: MCPeerID, peer: MCPeerID) {
     if session == nil {
-        session = transceiver.sessionForPeer(peer)
+        session = transceiver.session.mcSession
```
Comment (jpsim) [inline PeerKit/PeerKit.swift]: well done!

**Comment by jpsim** [comment]
Many many many thanks for this, @alexanderjarvis!

**Comment by alexanderjarvis** [comment]
@jpsim no problem :)

**Pull request merged**